opnieuw/retries.py

```diff
@@ -67,7 +70,7 @@ def calculate_exponential_multiplier(
     count = 2 ** (max_calls_total - 1) - 1
     multiplier = retry_window_after_first_call_in_seconds / max(count, 1)
 
-    return multiplier
+    return cast(float, multiplier)
```

Comment:
If we write `2.0` and `1.0` in the expressions above instead of `2` and `1`, will Mypy infer float automatically?

Reply:
Interestingly, one of `count = *2* ** (max_calls_total - *1*) - 1` has to be float literal for mypy to not complain.

Reply:
That makes sense, exponentiation between ints results in an int, if the base or the exponent is a float, you get float exponentiation:
```python
>>> type(2 ** 2)
<class 'int'>
>>> type(2.0 ** 2)
<class 'float'>
>>> type(2 ** 2.0)
<class 'float'>
```

What I don’t understand is why Mypy did not infer `multiplier` to float either way, because `/` between integers returns a float.

Reply:
Sure, and also a `-1.0` in the exponentiation would make it float, though mypy does not consider that either.